apollo-router-core/src/query_planner/mod.rs

```diff
@@ -89,89 +89,99 @@ impl PlanNode {
         schema: &'a Schema,
         parent_value: &'a Value,
     ) -> future::BoxFuture<(Value, Vec<Error>)> {
-        Box::pin(async move {
-            tracing::trace!("Executing plan:\n{:#?}", self);
-            let mut value;
-            let mut errors = Vec::new();
-
-            match self {
-                PlanNode::Sequence { nodes } => {
-                    value = parent_value.clone();
-                    for node in nodes {
+        Box::pin(
+            async move {
+                tracing::trace!("Executing plan:\n{:#?}", self);
+                let mut value;
+                let mut errors = Vec::new();
+
+                match self {
+                    PlanNode::Sequence { nodes } => {
+                        let sequence_span = tracing::info_span!("sequence");
+                        let _entered = sequence_span.enter();
```

Comment:
aren't we supposed to not do that when there's an await?
https://docs.rs/tracing/latest/tracing/span/struct.Span.html#in-asynchronous-code

Reply:
The reason why I'm allowed to do that, is because i use [in_current_span](https://docs.rs/tracing/latest/tracing/instrument/trait.Instrument.html#method.in_current_span) at line 113